src/main/scala/org/apache/spark/sql/sparkdl_stubs/UDFUtils.scala

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2017 Databricks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.sparkdl_stubs
+
+import java.util.ArrayList
+import scala.collection.JavaConverters._
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.{Column, Row, SparkSession, SQLContext}
+import org.apache.spark.sql.catalyst.expressions.{Expression, ScalaUDF}
+import org.apache.spark.sql.expressions.UserDefinedFunction
+import org.apache.spark.sql.types.DataType
+
+object UDFUtils extends Logging {
+  /**
+   * Register a UDF to the given SparkSession, so as to expose it in Spark SQL
+   * @param spark the SparkSession to which we want to register the UDF
+   * @param name registered to the provided SparkSession
+   * @param udf the actual body of the UDF
+   * @return the registered UDF
+   */
+  def registerUDF(spark: SparkSession, name: String, udf: UserDefinedFunction): UserDefinedFunction = {
+    def builder(children: Seq[Expression]) = udf.apply(children.map(cx => new Column(cx)) : _*).expr
+    spark.sessionState.functionRegistry.registerFunction(name, builder)
+    udf
+  }
+
+  /**
+   * Composite a set of UDFs as a single UDF
+   * @param ctx an active SQLContext
+   * @param name the name of the UDF
+   * @param judfs a seuqnce of UDFs (in Java's ArrayList)
+   * @return the registered UDF
+   */
+  def registerCompositeUDF(
+    ctx: SQLContext,
+    name: String,
+    judfs: ArrayList[UserDefinedFunction]): UserDefinedFunction = {
+
+    val udfs = judfs.asScala
+    val udf = PipelinedUDF(name, udfs.head, udfs.tail: _*)
+    logWarning(s"Registering composite udf $name -> $udf to session ${ctx.sparkSession}")
+    registerUDF(ctx.sparkSession, name, udf)
+  }
+
+  /**
+   * Register a UserDefinedfunction (UDF) as a composition of several UDFs.
+   * The UDFs must have already been registered
+   * @param ctx an active SQLContext
```

Comment:
Since 2.x, we do not encourage users to use SQLContext. Instead, you can pass SparkSession.